### DISCUSSION TITLE
Writing a custom injection for specific language

### DISCUSSION THREAD

**Discussion by wingyplus** [zed-industries/zed#14953] [Feature Requests]
Is it possible to write a custom injection language to another language? For example, in Elixir, I want to injecting Markdown syntax into `moduledoc` to highlighting documentation in the code.

**Comment by everdrone** [comment]
Also interested in overriding language syntaxes with custom highlighting

**Comment by artegoser** [comment]
https://zed.dev/docs/extensions/languages#code-injections

**Reply by everdrone** [reply]
What about injections for multiple languages in a single extension? is it possible?

**Reply by artegoser** [reply]
It's working fine for me
![image](https://github.com/user-attachments/assets/ad94f989-aa7b-49c6-84df-2f69fed2fa4f)
![image](https://github.com/user-attachments/assets/861d6860-f149-4905-b6be-2ed4f14384a1)

**Reply by artegoser** [reply]
Or maybe I misunderstood. You should probably just create a subdirectory for each language in the languages folder and write injections.scm there. If you think about it logically, it should work

**Reply by everdrone** [reply]
I wanted to create an extension to highlight `TODO` comments, not for one specific language but virtually all supported ones
It's cumbersome to implement a `config.toml` and a different `overrides.scm` for each language supported by zed, instead having a global `overrides.scm` would pretty much solve the problem instantly

**Reply by kraanzu** [reply]
@everdrone, DId you find a solution to this?
I want to custom inject a diff language inside python but adding a `python/` directory inside `languages/` just overwrites the default highlighting in python

**Reply by everdrone** [reply]
@kraanzu I didn't try anymore. I don't know if things changed in the meantime but afaik it's still not possible

**Reply by JonGretar** [reply]
> @everdrone, DId you find a solution to this? I want to custom inject a diff language inside python but adding a `python/` directory inside `languages/` just overwrites the default highlighting in python

You would have to fork the python language and install as dev extension.

**Reply by kraanzu** [reply]
Yea I dug a bit more and seems like this is not possible in zed rn. Other editors such as nvim and helix gives an option to inherit the parent language config but zed has fundamentally different code design from what I read from the comments.

**Comment by itahol** [comment]
Hey, in my job I work in typescript, where we use both an ORM and raw queries for our DB.
I would like to have a go at injecting SQL in the raw queries strings, and have it be done based on the content of the string and not some kind of tagging (so for example match a regex of "select .* from .*" for keeping it simple).

I could add it to the language injections file, but I think it's not something that should be in the base language injections.scm file as it might not be something that everyone is interested in.
Therefore, it seems like a good case for an extension :)

Is it possible to add to a language injections without registering a new grammer/language?
I couldn't really understand from the docs, if someone has experience with it I would appreciate it!

I'm currently AFK but when I have the chance I'll try writing it naively and see if it works.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
